components/org.wso2.carbon.identity.conditional.auth.functions.user.store/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/store/UserStoreFunctions.java

```diff
@@ -55,15 +57,17 @@ public JsAuthenticatedUser getUniqueUserWithClaimValues(Map<String, String> clai
             return null;
         }
         if (parameters.length == 2) {
-            if ( parameters[0] instanceof String) {
-                tenantDomain = (String) parameters[0];
+            if ( parameters[0] instanceof JsAuthenticationContext) {
+                authenticationContext = (JsAuthenticationContext) parameters[0];
+                tenantDomain = authenticationContext.getContext().getTenantDomain();
```

Comment:
Isn't it better to keep the InstanceOf String check in the else block as well?

Reply:
Keeping first param as tenant domain causes NPE, that's why I removed it from public